### DISCUSSION TITLE
Hiding collections that the current user has no access to

### DISCUSSION THREAD

**Discussion by websmurf** [sulu/sulu#6903] [Q&A]
I'm currently trying to restrict access to certain collections for certain users. 

For this reason I've enabled permissions on a few collections. The effect of this, that a permission icon will be displayed on the folder and that the users who can't access them won't be able to see the collection content. So far so good :-)

I would like to hide the collection entirely for users who can't access them. Is there a way to achieve this?
Or perhaps a way to lock a user to a certain collection (in stead of the root collection) so that he can only see the sub-collections of that collection.

I've found a [Stackoverflow](https://stackoverflow.com/questions/65500023/hide-sulu-media-collection-tiles-when-permissions-dont-allow-viewing-of-that-co) post from a while back with the same question, but no resolution or solution unfortunately.

**Comment by NeuralClone** [comment]
I'm not sure if this can be achieved with how permissions currently work since you need to grant both the edit and view permissions to every role that should have access to the Media section. If you deny a role those permissions, it won't be able to access Media at all.

What I've done is granted restricted roles the view and edit permissions for Media. Then I've then taken away permissions for each collection the role shouldn't have access to. The end result shows a list of the root collections that you've created. If you try to access any collection you don't have permission to view, it'll deny access to everything inside it. It'll show the number of items the root collections contain, but it won't show you what those items are if you try to access the collection.

So the basic idea is you grant a role the maximum amount of access you want it to have and then take permissions away from it for any collection it shouldn't be able to access. It sounds like you might already be doing this. Just make sure you deny the security permission to these restricted roles.

This isn't ideal since you're still revealing some of the collection names. But I'm not sure how else to do this without reworking parts of Sulu. If there's another way to do this with how things are currently written, I haven't been able to figure out what it might be.

For example:

Settings -> User roles -> Second Admin
![Screenshot 2023-01-16 173538](https://user-images.githubusercontent.com/678221/212775956-3d8df5e4-2858-4c82-8b87-5c781e58756f.png)

Permissions for a collection the role can access
![Screenshot-2023-01-16-175633b](https://user-images.githubusercontent.com/678221/212777248-85548aca-0134-43ae-bb22-a6b691ff02c0.png)

Permissions for a restricted collection (Website permissions aren't necessary - kept here just as a possible setup)
![Screenshot-2023-01-16-173608](https://user-images.githubusercontent.com/678221/212775969-ffb7c9c9-3a71-4e1b-8b52-80dbc7d8eecf.png)

**Comment by websmurf** [comment]
@NeuralClone I have that setup pretty much the same.

My only additional wish based on what is currently possible, is that the folders that are currently displayed with a lock icon, can be hidden.

**Comment by websmurf** [comment]
Alright, after a lot of searching I found a small hack that can be applied in the `assets/admin/node_modules/sulu-admin-bundle/containers/List/adapters/FolderAdapter.js` file to get the behaviour I want (so a folder that has permissions enabled and cannot be viewed will be filtered out):

Change this line (around line 46):
```
{data.map((item: Object) => (
```

to
```
{data.filter(item => item._hasPermissions === false || item._permissions.view === true).map((item: Object) => (
```

Ofcourse you would have to create a custom build of the admin panel for this to work:
https://docs.sulu.io/en/2.5/cookbook/build-admin-frontend.html

I still looking for a better way to apply server side filtering, but haven't found a clean solution yet.
